controller/eventGroupController.js

```diff
@@ -0,0 +1,172 @@
+const HttpStatus = require("http-status-codes");
+
+const EventGroup = require("../model/eventGroupModel");
+const Event = require('../model/eventModel');
+const ValidationUtil = require('../util/validationUtil');
+const StatusEnum = require('../model/enums').StatusEnum;
+
+
+// =================================== CRUD FUNCTIONS ===================================
+
+
+const create = (req, res) => {
+    const loggedUser = req.decoded;
+
+    if (!loggedUser || !req.body) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+    // TODO: AdminAuth MW
+    // TODO: add new user to eventGroup (update eventGroup.users)
+    // TODO: remove user from eventGroup
+    const eventGroup = new EventGroup(req.body);
+
+    if (loggedUser.companyID != eventGroup.companyID) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized'
+        });
+    }
+
+    // Check if event exists 
+    Event.findOne({ _id: eventGroup.eventID, companyID: eventGroup.companyID }).then(event => {
```

Comment:
Should you not check if the IDs are valid first? ``ValidationUtil.isValidObjectId()`. I seem to recall you saying that if the IDs weren't valid an Error is thrown that is not caught inside the `catch()` function.